DOC: batching in 1D/ND interpolation/smoothing routines (#22531)
scipy/scipy: doc/source/tutorial/interpolate/smoothing_splines.rst

result, ``lam=None``, is close to the underlying sine curve.

.. _tutorial-interpolate_GCV_batching:

> Batching of `y` arrays

Reviewer: ```suggestion
Batching of ``y`` arrays
```

> ``````````````````````

Author: ```suggestion
````````````````````````
```

> 

Author: Yeah, made the change locally and pushed.

`make_smoothing_spline` constructor accepts multidimensional ``y`` arrays and an

> optional `axis=0` parameter and interprets them exactly the same way the

Reviewer: ```suggestion
optional ``axis`` parameter and interprets them exactly the same way the
```

   >>> plt.show()

> Batching of `y` arrays

Reviewer: ```suggestion
Batching of ``y`` arrays
```

----------------------

Unlike :ref:`interpolating splines <tutorial-interpolate_batching>` and
:ref:`GCV smoothers <tutorial-interpolate_GCV_batching>`, `make_splrep` and `make_splprep`
do **not** allow multidimensional ``y`` arrays and require that ``x.ndim == y.ndim == 1``.

> The technical reason for this limitation is that the knot vector depends on the ``y``
> values, thus for a batched ``y``, the ``t`` batch could be a ragged array, which `BSpline`
> is not equipped to handle.

Reviewer: ```suggestion
The technical reason for this limitation is that the length of the knot vector ``t``
depends on the ``y`` values, thus for a batched ``y``, the batched ``t`` could be a
ragged array, which `BSpline` is not equipped to handle.
```

Therefore if you need to handle batched inputs, you will need to loop over the
batch manually and construct a `BSpline` object per slice of the batch.
Having done that, you can mimic `BSpline` behavior for evaluations with a workaround
along the lines of

.. code-block:: python

    class BatchSpline:

>         """BSpline-line class with reference batch behavior."""

Reviewer: ```suggestion
        """BSpline-like class with batch behavior."""
```
"reference" was used in the context of the tests, but probably isn't needed here.
(and "line" was a typo; I meant "like")

>         def __init__(self, x, y, axis, *, spline, **kwargs):

Reviewer: I think the name `spline` was good enough for the tests, but is the meaning of the name obvious enough here?

Author: I think it's fine. The class name is `BatchSpline`, the docstring says it implements a "BSpline-like functionality", and its argment is named `spline`,  which is then used as a callable. It all squares out, AFAICS.
